beta-src/src/models/TerritoryClass.ts

```diff
@@ -0,0 +1,72 @@
+import { ITerritory, IBorder, ICoastalBorder } from "./Interfaces";
+import UnitClass from "./UnitClass";
+
+// We may not need this. Since almost all getter methods require broader data from the entire board and setter methods can be done when we create this object.
+
+export default class Territory {
+  id: number;
+
+  name: string;
+
+  type: string;
+
+  supply: boolean;
+
+  countryID: number;
+
+  coast: string;
+
+  coastParentID: number;
+
+  smallMapX: number;
+
+  smallMapY: number;
+
+  Borders: IBorder[];
+
+  CoastalBorders: ICoastalBorder[];
+
+  convoyLink: boolean;
+
+  coastParent: Territory;
+
+  Unit: UnitClass;
+
+  unitID: number;
+
+  constructor({
+    id,
+    name,
+    type,
+    supply,
+    countryID,
+    coast,
+    coastParentID,
+    smallMapX,
+    smallMapY,
+    Borders,
+    CoastalBorders,
+    coastParent,
+    Unit,
+    unitID,
+  }: ITerritory) {
+    this.id = id;
+    this.name = name;
+    this.type = type;
+    this.countryID = countryID;
+    this.coast = coast;
+    this.coastParentID = coastParentID;
+    this.smallMapX = smallMapX;
+    this.smallMapY = smallMapY;
+    this.Borders = Borders;
+    this.CoastalBorders = CoastalBorders;
+    this.coastParent = new Territory(coastParent);
+    this.Unit = Unit;
+    this.unitID = unitID;
```

Comment:
for every thing where you are declaring a property that is something passed to the constructor you can use this pattern to achieve the same thing, but with much less code:

```
class Territory {
  constructor(public id: number, public name: string) {}
}
```

this does the same thing as:

```
class Territory {
  id: number;
  name: string;
  constructor(id, name) {
    this.id = id;
    this.name = name;
  }
}
```

Reply:
Some, I was able to do it this way but some I could not. Looks like TS is not happy with using parameter properties with binding patterns.